src/control/server/harness.go

```diff
@@ -104,6 +104,25 @@ func (h *IOServerHarness) GetManagementInstance() (*IOServerInstance, error) {
 	return h.instances[defaultManagementInstance], nil
 }
 
+// GetMSLeaderInstance returns a managed IO Server instance to be used as a
+// management target and fails if selected instance is not MS Leader.
+func (h *IOServerHarness) GetMSLeaderInstance() (*IOServerInstance, error) {
+	h.RLock()
```

Comment:
Is there a reason why you place the rwlock first in this struct but the mutex at the bottom in the client connection. If you put the client connection one up top could you just do h.Lock and h.Unlock on the mutex there as well?

Reply:
yes good point, have changed it to use an embedded type as you suggest